tests/phpunit/CRM/Core/BAO/SettingTest.php

```diff
@@ -180,4 +180,30 @@ public static function _testOnChange_onChangeExample($oldValue, $newValue, $meta
     $_testOnChange_hookCalls['metadata'] = $metadata;
   }
 
+  /**
+   * Test to set civicrmEnvironment
+   *
+   */
+  public function testSetCivicrmEnvironment() {
+    CRM_Core_BAO_Setting::setItem(TRUE, CRM_Core_BAO_Setting::DEVELOPER_PREFERENCES_NAME, 'civicrmEnvironment');
+    $values = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::DEVELOPER_PREFERENCES_NAME, 'civicrmEnvironment');
+    $this->assertEquals(TRUE, $values);
+
+    global $civicrm_setting;
+    $civicrm_setting[CRM_Core_BAO_Setting::DEVELOPER_PREFERENCES_NAME]['civicrmEnvironment'] = FALSE;
+    Civi::service('settings_manager')->useMandatory();
+    $values = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::DEVELOPER_PREFERENCES_NAME, 'civicrmEnvironment');
+    $this->assertEquals(FALSE, $values);
+
```

Comment:
Please add a defaults test similar to 115 to ensure that default is production=true.